tests/arm/checks/resource/test_AppGWDefinesSecureProtocols.py

```diff
@@ -0,0 +1,43 @@
+import os
+import unittest
+from checkov.runner_filter import RunnerFilter
+from checkov.arm.runner import Runner
+from checkov.arm.checks.resource.AppGWDefinesSecureProtocols import check
+
+
+class TestAppGWDefinesSecureProtocols(unittest.TestCase):
+
+    def test(self):
+        runner = Runner()
+        current_dir = os.path.dirname(os.path.realpath(__file__))
+        test_files_dir = os.path.join(current_dir, "example_AppGWDefinesSecureProtocols")
+        report = runner.run(root_folder=test_files_dir, runner_filter=RunnerFilter(checks=[check.id]))
+        summary = report.get_summary()
+
+        passing_resources = {
+            'Microsoft.Network/applicationGateways.pass',
+            'Microsoft.Network/applicationGateways.pass1'
+        }
+        failing_resources = {
+            'Microsoft.Network/applicationGateways.fail',
+            'Microsoft.Network/applicationGateways.fail1',
+            'Microsoft.Network/applicationGateways.fail2',
+            'Microsoft.Network/applicationGateways.fail3',
+        }
```

Comment:
Maybe we should give the resources a better name to indicate what this resource should be, like fail_with_cipher or something like that.